Examples/Scripts/ApplyAutoRecleanPipelineBatch.sh

```diff
@@ -0,0 +1,178 @@
+# Global Output Variables
+#	${StudyFolder}			- Path to folder containing all subjects data in subdirectories named 
+#							  for the subject id
+#	${Subjlist}				- Space delimited list of subject IDs
+#	${EnvironmentScript}	- Script to source to setup pipeline environment
+#	${RunLocal}				- Indication whether to run this processing "locally" i.e. not submit
+#							  the processing to a cluster or grid
+#
+get_options() {
+	local scriptName=$(basename ${0})
+	local arguments=("$@")
+
+	# initialize global output variables
+	StudyFolder="${DEFAULT_STUDY_FOLDER}"
+	Subjlist="${DEFAULT_SUBJECT_LIST}"
+	EnvironmentScript="${DEFAULT_ENVIRONMENT_SCRIPT}"
+	RunLocal="${DEFAULT_RUN_LOCAL}"
+
+	# parse arguments
+	local index=0
+	local numArgs=${#arguments[@]}
+	local argument
+
+	while [ ${index} -lt ${numArgs} ]
+	do
+		argument=${arguments[index]}
+
+		case ${argument} in
+			--StudyFolder=*)
+				StudyFolder=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--Subject=*)
+				Subjlist=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--EnvironmentScript=*)
+				EnvironmentScript=${argument#*=}
+				index=$(( index + 1 ))
+				;;
+			--runlocal | --RunLocal)
+				RunLocal="TRUE"
+				index=$(( index + 1 ))
+				;;
+			*)
+				echo "ERROR: Unrecognized Option: ${argument}"
+				exit 1
+				;;
+		esac
+	done
+
+	# check required parameters
+	if [ -z ${StudyFolder} ]
+	then
+		echo "ERROR: StudyFolder not specified"
+		exit 1
+	fi
+
+	if [ -z "${Subjlist}" ]
+	then
+		echo "ERROR: Subjlist not specified"
+		exit 1
+	fi
+
+	if [ -z ${EnvironmentScript} ]
+	then
+		echo "ERROR: EnvironmentScript not specified"
+		exit 1
+	fi
+
+	if [ -z ${RunLocal} ]
+	then
+		echo "ERROR: RunLocal is an empty string"
+		exit 1
+	fi
+
+	# report options
+	echo "-- ${scriptName}: Specified Command-Line Options: -- Start --"
+	echo "   StudyFolder: ${StudyFolder}"
+	echo "   Subjlist: ${Subjlist}"
+	echo "   EnvironmentScript: ${EnvironmentScript}"
+	echo "   RunLocal: ${RunLocal}"
+	echo "-- ${scriptName}: Specified Command-Line Options: -- End --"
+}  # get_options()
+
+#
+# Function Description
+#	Main processing of this script
+#
+#	Gets user specified command line options and runs a batch of ReApplyFix processing
+#
+main() {
+	# get command line options
+	get_options "$@"
+
+	# set up pipeline environment variables and software
+	source "$EnvironmentScript"
+
+	# set list of fMRI
+	fMRINames="rfMRI_REST1_RL@rfMRI_REST1_LR@rfMRI_REST2_LR@rfMRI_REST2_RL"
+  
+	# specify the name of concatenated folder
+	# if run Multi-Run specify ConcatNames as null string
+	MRConcatfMRIName="rfMRI_REST"
+
+	# set highpass
+	highpass=2000
+	
+	# set resolution
+	fMRIResolution="2"
+
+	# set singularity path
+	PythonSingularity="/path/to/singularity.img"
+	PythonSingularityMountPath="/path/to/data" # where the data dir need to mount to the singularity
+
+	# # set python interpreter
+	# PythonInterpreter="/my/conda/path/envs/hcp_python_env/bin/python3"
+	PythonInterpreter=""
```

Comment:
We may want to expand these settings to two sections, one for singularity, and one for conda, like so:

```
	#run in singularity
	PythonSingularity="/path/to/singularity.img"
	PythonSingularityMountPath="/path/to/data" # where the data dir need to mount to the singularity
	PythonInterpreter=""

	#run in conda
	#PythonSingularity=""
	#PythonSingularityMountPath=""
	#PythonInterpreter="/my/conda/path/envs/hcp_python_env/bin/python3"